### PR TITLE
feat: Context succeeding - Part 6 - topic-operator mockkube kafka-init api

### PR DESCRIPTION
**Pull request by samuel-hawker**
For this Pull request I have continued the context
succeeding work, and taken the opportunity to remove
unneeded concurrency based dependencies.
minor test renames, hamcrest additions and variable
renames to improve clarity.

This contains improvements to the topic-operator,
mockkube, kafka-init and api sub-modules.
I put some work into removing the passing of Exceptions
in test code and instead to error immediately out of tests
on fatal throws.

Contributes to: #2647

Signed-off-by: Samuel Hawker samuel.hawker@ibm.com

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

File: mockkube/src/test/java/io/strimzi/test/io/strimzi/test/mockkube/MockKubeTest.java [hunk 1]
```diff
@@ -182,55 +179,51 @@ public void podNameScopedCreateListGetDelete(Class<RT> cls,
...
+        assertThat(items, hasSize(1));
         RT item = items.get(0);
         item.getMetadata().setResourceVersion(null);
         item.getMetadata().setGeneration(null);
         assertThat(item, is(pod));
 
         // List with namespace
         items = mixedOp.apply(client).inNamespace("other").list().getItems();
-        // TODO assertEquals(0, items.size());
+        // TODO assertThat(items.size(), is(0));
```
[the remark this answers is not in the document]
Comment (samuel-hawker) [reply mockkube/src/test/java/io/strimzi/test/io/strimzi/test/mockkube/MockKubeTest.java]: I thought exactly this as well - I uncommented it out and the line fails.
My understanding is that this has been left commented out as a reminder that this _should_ be passing but isn't...
I.e. `mixedOp.apply(client).inNamespace("other").*` should actually be behaving differently to how it currently is. (Incomplete work in the MockKube)
Similar for this other line in the same file:
https://github.com/strimzi/strimzi-kafka-operator/blob/abb757cf33c430e8925152f77c5c78102534735d/mockkube/src/test/java/io/strimzi/test/io/strimzi/test/mockkube/MockKubeTest.java#L236

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=acceptance testcase=io.strimzi.systemtest.ConnectST#testKafkaConnectAndConnectorFileSinkPlugin

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=acceptance testcase=io.strimzi.systemtest.ConnectST#testKafkaConnectAndConnectorFileSinkPlugin

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**